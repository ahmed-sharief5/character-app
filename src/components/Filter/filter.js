import React from "react";
import PropTypes from 'prop-types';

const Filter = ({ elements, value, kind, onSelect }) => {
    return (
        <div className="dropdown">
            <button className="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
                {value ? value : kind}
            </button>
            <div className="dropdown-menu" aria-labelledby="dropdownMenuButton">
                {elements && elements.length > 0 ? 
                    elements.map((ele, idx) => {
                        return <a key={idx} className="dropdown-item" onClick={onSelect} href="#">{ele}</a>
                    })
                    : <a className="dropdown-item" href="#">No items</a>
                }
            </div>
        </div>
    );
}

Filter.propTypes = {
    elements: PropTypes.array,
    value: PropTypes.string,
    kind: PropTypes.string,
    onSelect: PropTypes.func,
};

export default Filter;