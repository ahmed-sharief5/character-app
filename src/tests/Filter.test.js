import React from "react";
import Filter from "../components/Filter";
import renderer from "react-test-renderer";

it("renders filter component correctly", () => {
    const items = ['one', 'two', 'three'];
    const tree = renderer.create(<Filter elements={items}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
