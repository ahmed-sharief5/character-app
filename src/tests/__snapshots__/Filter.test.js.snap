// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders filter component correctly 1`] = `
<div
  className="dropdown"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    className="btn btn-secondary dropdown-toggle"
    data-toggle="dropdown"
    id="dropdownMenuButton"
    type="button"
  />
  <div
    aria-labelledby="dropdownMenuButton"
    className="dropdown-menu"
  >
    <a
      className="dropdown-item"
      href="#"
    >
      one
    </a>
    <a
      className="dropdown-item"
      href="#"
    >
      two
    </a>
    <a
      className="dropdown-item"
      href="#"
    >
      three
    </a>
  </div>
</div>
`;
